guides/plugins/plugins/content/cms/real-world-cms/add-real-world-cms-block.md

```diff
@@ -0,0 +1,268 @@
+       We give the slot button the element text in this tutorial because we haven't defined a button element yet.
+       You can use the existing blocks and elements out of the box but it gets interesting if you want to create your very own element with an own config.
+       We'll take a look at this later in this guide.
+    */
+    slots: {
+        button: 'text'
+    }
+});
+```
+{% endcode %}
+
+
+### Block component
+
+Luckily we are just creating a basic button in this guide so the code should be good to understand how everything works.
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/component/sw-cms-block-button.html.twig" %}
+{% raw %}
+```text
+{% block sw_cms_block_button %}
+	<div class="sw-cms-block-button">
+		<slot name="button"></slot>
+	</div>
+{% endblock %}
+```
+{% endraw %}
+{% endcode %}
+
+We also need to register the component:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/component/index.js" %}
+```javascript
+import template from './sw-cms-block-button.html.twig';
+
+const { Component } = Shopware;
+
+Component.register('sw-cms-button', {
+    template
+});
+```
+{% endcode %}
+
+
+### Block preview
+
+We'll keep the preview very basic in this guide like this:
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/preview/sw-cms-preview-button.html.twig" %}
+{% raw %}
+```text
+{% block sw_cms_block_ninja_button_preview %}
+	<div class="ninja-button-flex-center">
+		<button class="sw-cms-preview-ninja-button">Button</button>
+	</div>
+{% endblock %}
+```
+{% endraw %}
+{% endcode %}
+
+Let's register this component as well like this:
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/preview/index.js" %}
+{% raw %}
+```text
+import template from './sw-cms-preview-button.html.twig';
+import './sw-cms-preview-button.scss';
+
+const { Component } = Shopware;
+
+Component.register('sw-cms-preview-button', {
+    template
+});
+```
+{% endraw %}
+{% endcode %}
+
+And let's add a bit of CSS code (not neccessary)
+
+{% code title="<plugin root>/src/Resources/app/administration/src/module/sw-cms/blocks/text/cms-button/preview/sw-cms-preview-button.scss" %}
+```css
+.ninja-button-flex-center {
+    display: flex;
+    align-items: center;
+    justify-content: center;
+    padding: 1rem;
+    .sw-cms-preview-ninja-button {
+        background-color: #4492ed;
+        border: none;
+        color: white;
+        padding: 16px 60px;
+        text-align: center;
+        text-decoration: none;
+        display: inline-block;
+        font-size: 16px;
+    }
+
+}
+```
+{% endcode %}
+
+### Build administration
+
+You need to build your administration before you can see anything in the shopware backend. Just run the one of the following commands:
+
+{% tabs %}
+{% tab title="Development template" %}
```

Comment:
we're trying to move away from the Development template, so I'd recommend running 
`composer run build:js:admin` instead

Reply:
I think `composer run build:js:admin` is not running on every environment out of the box? So probably would make sense to also see the other options right?

Reply:
@Christian-Rades Isn't this a topic someone has to normalize for all scripts anyways with deprecation of shopware/development ? Maybe it is good to keep it the same over the same page so it is easier to normalize later.